pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -2357,6 +2368,78 @@ func (l *LibvirtDomainManager) parseFSDisks(fsDisks []api.FSDisk) []v1.VirtualMa
 	return disks
 }
 
+func (l *LibvirtDomainManager) linkImageVolumeFilePaths(vmi *v1.VirtualMachineInstance) error {
+	for diskIndex, volume := range vmi.Spec.Volumes {
+		if volume.ContainerDisk == nil {
+			continue
+		}
+		backingFile := containerdisk.GetDiskTargetPathFromLauncherView(diskIndex)
+		fileToSoftLink, err := getDiskTargetPathFromImageVolumeView(diskIndex, volume.ContainerDisk.Path)
+		if err != nil {
+			return fmt.Errorf("failed to find disk file from ImageVolume: %v", err)
+		}
+		err = os.Symlink(fileToSoftLink, backingFile)
+		if err != nil && !os.IsExist(err) {
+			return fmt.Errorf("error creating symlink for containerDisk: %v", err)
+		}
+	}
+
+	firmware := vmi.Spec.Domain.Firmware
+	if firmware == nil {
+		return nil
+	}
+
+	if kutil.HasKernelBootContainerImage(vmi) {
+		kb := firmware.KernelBoot
+
+		err := os.MkdirAll(containerdisk.GetKernelBootArtifactPathFromLauncherView(""), 0755)
+		if err != nil {
+			return fmt.Errorf("error creating dir for KernelPath: %v", err)
+		}
+		log.Log.Object(vmi).Infof("kernel boot defined for VMI. Converting to domain XML")
+		if kb.Container.KernelPath != "" {
+			kernelPath := containerdisk.GetKernelBootArtifactPathFromLauncherView(kb.Container.KernelPath)
+			fileToSoftLink := getKernelBootArtifactPathFromImageVolumeView(kb.Container.KernelPath)
+			err := os.Symlink(fileToSoftLink, kernelPath)
+			if err != nil && !os.IsExist(err) {
+				return fmt.Errorf("error creating symlink for KernelPath: %v", err)
+			}
+		}
+
+		if kb.Container.InitrdPath != "" {
+			initrdPath := containerdisk.GetKernelBootArtifactPathFromLauncherView(kb.Container.InitrdPath)
+			fileToSoftLink := getKernelBootArtifactPathFromImageVolumeView(kb.Container.InitrdPath)
+			err := os.Symlink(fileToSoftLink, initrdPath)
+			if err != nil && !os.IsExist(err) {
+				return fmt.Errorf("error creating symlink for KernelPath: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func getDiskTargetPathFromImageVolumeView(volumeIndex int, volumePath string) (string, error) {
```

Comment:
from image volume view?
what do you mean here? is it from the host's of launcher's view?